Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -14,311 +14,403 @@
 //
 //******************************************************************************
 
-#include "Starboard.h"
+#import <Foundation/NSException.h>
+#import <NSOrderedSetInternal.h>
+#import <Starboard.h>
 #import <StubReturn.h>
-
-#include "Foundation/NSOrderedSet.h"
+#import <VAListHelper.h>
 
 @implementation NSOrderedSet
+
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSet {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[NSOrderedSet alloc] init] autorelease];
```

Comment:
the + convenience constructors should call [self alloc] instead of [NSOrderedSet alloc], otherwise this code won't correctly alloc for NSMutableOrderedSet or other subclasses